### ISSUE TITLE
BitmapImage doesn't use logical pixels when decoding a stream

### ISSUE THREAD

**Issue by FrayxRulez** [microsoft/microsoft-ui-xaml#6257]
### Describe the bug

Currently, setting `DecodePixelType = DecodePixelType.Logical` doesn't seem to make any difference in a WinUI 3 desktop application.

### Steps to reproduce the bug

1. Set the monitor display to a scaling that's not 100%.
2. Create a new BitmapImage, specifying `DecodePixelWidth` and `DecodePixelHeight`.
3. `DecodePixelType = DecodePixelType.Logical`.
4. Display the bitmap in the UI.
5. Notice the blur.

### Expected behavior

BitmapImage will scale decode pixel width and height accordingly to the monitor scaling when pixel type is set to logical.

### Screenshots

Here's a comparison between UWP on the left and WinUI 3 on the right:
![image](https://user-images.githubusercontent.com/8152849/140807180-d25d2b77-9ae2-4bb1-ac41-82e04e243687.png)

### NuGet package version

WinUI 3 - Windows App SDK 0.8: 0.8.0 (If you're seeing your issue in older previews of WinUI 3, please try this release)

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Insider Build (xxxxx)

**Comment by DL444** [comment]
Still experiencing this problem in Windows App SDK v1.1.2.

Also note that `DisplayInformation` APIs are currently not available in Windows App SDK since they require a `CoreWindow`. See https://github.com/microsoft/WindowsAppSDK/issues/1609.

I eventually resorted to calling Win32 APIs directly to get the scaling factor and multiply it with the desired logical pixel size to get the corresponding physical pixel size, and then assign that to `DecodePixelWidth`. I'll post my code here for reference.

```cs
using System;
using System.Runtime.InteropServices;
using WinRT.Interop;

public static double GetWindowScaleFactor(Window window)
{
    IntPtr hwnd = WindowNative.GetWindowHandle(window);
    IntPtr hMon = MonitorFromWindow(hwnd, 1);
    int hresult = GetScaleFactorForMonitor(hMon, out int scale);
    return hresult == 0 ? scale / 100.0 : 0;
}

[DllImport("User32.dll")]
private static extern IntPtr MonitorFromWindow(IntPtr hwnd, uint dwFlags);

[DllImport("Shcore.dll")]
private static extern int GetScaleFactorForMonitor(IntPtr hMon, out int pScale);
```

**Comment by Marv51** [comment]
This issue is still present in WinAppSDK 1.2.2. 

`DecodePixelType.Logical` is basically just ignored. 

This results in DecodePixelWidth/Height being very inconvenient use.

Seems like this was deactivated during the transition to 3.0 and not fixed afterwards?

I wonder if this is tracked internally? I wonder who would know, maybe @codendone?

**Comment by DominicMaas** [comment]
I've hit this as well. Giving it a bump.

**Comment by bpulliam** [comment]
This issue should be fixed in WinAppSDK 1.3.

**Comment by JJBrychell** [comment]
It looks like the fix may not  have made it into 1.3, but it is in1.4.
